File: 2021/01/day1_part1.py
num_increased = 0

with open("input.txt") as f:
    prev = None
    for line in f:
        if prev == None:
            print("{} (N/A - No previous measurement)".format(line))
            prev = int(line)
            continue
        if int(line) > prev:
            num_increased += 1
            print("{} (increased)".format(line.strip()))
        else:
            print("{} (decreased)".format(line.strip()))
        prev = int(line)

print("Number of increases: {}".format(num_increased))

File: 2021/01/day1_part1.ts
// NOTE: Using Deno (v1.16.3), not Node

try {
	let numIncreased = 0
	const inputText = await Deno.readTextFile('input.txt')
	let prev: number | undefined
	inputText.split('\n').forEach(line => {
		let val = parseInt(line)
		if (!prev) {
			prev = val
			return
		}
		if (val > prev) {
			numIncreased++
		}
		prev = val
	})
	console.log(`Number of increases: ${numIncreased}`)
} catch (error) {
	console.error(error)
}

